primary/src/tests/proposer_tests.rs

```diff
@@ -73,15 +81,66 @@ async fn propose_payload() {
         metrics,
     );
 
-    // Send enough digests for the header payload.
+    // Ensure the proposer does not make an empty proposal
+    tokio::select! {
+        _result = rx_headers.recv() => panic!("Unexpected header proposal"),
+        () = sleep(delta * 10) => ()
+    }
+
+    // Now send enough digests for the header payload.
     let name_bytes: [u8; 32] = *name.0.as_bytes();
     let digest = BatchDigest(name_bytes);
     let worker_id = 0;
     tx_our_digests.send((digest, worker_id)).await.unwrap();
 
-    // Ensure the proposer makes a correct header from the provided payload.
+    // Ensure the proposer makes a valid proposal
     let header = rx_headers.recv().await.unwrap();
     assert_eq!(header.round, 1);
     assert_eq!(header.payload.get(&digest), Some(&worker_id));
     assert!(header.verify(&committee(None)).is_ok());
 }
+
+#[tokio::test]
+async fn advance_to_help() {
```

Comment:
I would like a test to explore what happens when most nodes submit empty headers as well.